src/opnsense/service/templates/OPNsense/IDS/suricata.yaml

```diff
@@ -254,6 +254,9 @@ outputs:
             # output TLS transaction where the session is resumed using a
             # session id
             session-resumption: {{ 'no' if helpers.empty('OPNsense.IDS.general.eveLog.tls.sessionResumption') else 'yes' }}
+            # ja4 hashes in tls records will never be logged unless
+            # the following is set to on. (Default off)
+            ja4: {{ 'on' if helpers.empty('OPNsense.IDS.general.eveLog.tls.custom') or 'ja4' in OPNsense.IDS.general.eveLog.tls.custom else 'off' }}
```

Comment:
this defaults to `on` (empty custom property)

Reply:
Updated to be enabled if TLS extended logging is enabled or JA4 is explicitly listed in the custom fields given JA4 is considered a default field part of the TLS extended logging: https://github.com/OISF/suricata/blob/66e47a1983bee06fb85418d8ca053d0d9a372474/src/output-json-tls.c#L101-L114

Reply:
@0xThiebaut ok, next question, where is `ja4` specified in suricata for tls? I couldn't find it in their reference configuration or their docs myself.

https://github.com/OISF/suricata/blob/66e47a1983bee06fb85418d8ca053d0d9a372474/suricata.yaml.in#L281-L288

https://docs.suricata.io/en/latest/output/eve/eve-json-output.html#tls

Reply:
@0xThiebaut ok, that's confusing, latest isn't "latest" apparently in the documentation, thanks for clearing that up :)